### PR TITLE
Option to disable logging of log producer (2nd try)

### PR DESCRIPTION
**Pull request by marcstern**
[Issue #1069 - Marc Stern]

### CODE REVIEW

**Comment by marcstern** [comment]
Fixed

File: apache2/msc_logging.c [hunk 1]
```diff
@@ -1271,6 +1271,9 @@ void sec_audit_logger_json(modsec_rec *msr) {
         if (been_opened == 1) {
             yajl_gen_map_close(g); // sanitized args map is finished
         }
+#ifdef LOG_NO_PRODUCER
```
Comment (zimmerle) [inline apache2/msc_logging.c]: This is way more than the simple producer tag.

File: apache2/msc_logging.c [hunk 2]
```diff
@@ -2013,6 +2013,9 @@ void sec_audit_logger_native(modsec_rec *msr) {
             sec_auditlog_write(msr, text, strlen(text));
         }
 
+#ifdef LOG_NO_PRODUCER
```
Comment (zimmerle) [inline apache2/msc_logging.c]: Same as above - too many things ignored that are not the producer.

**Comment by marcstern** [comment]
The goal was to remove all the "contextual" environment, which is indeed more than the producer. Would it be OK with an explicit name (also in configure) like "LOG_NO_SERVER_CONTEXT"?

**Comment by zimmerle** [comment]
If that is the goal, it should be clear to whom may read the code and to the user who may enabling it during the configure. So we have to change the help message and the if/defs.

**Comment by marcstern** [comment]
Done: {dis|en}able-server-context-logging

**Pull request closed without merging**